featurebyte/query_graph/node/metadata/sdk_code.py

```diff
@@ -478,6 +509,43 @@ def generate(self, to_format: bool = False) -> str:
             elif isinstance(statement, str):
                 statement_lines.append(statement)
         statements = "\n".join(statement_lines)
+        return statements, import_pairs
+
+    def generate(
+        self,
+        to_format: bool = False,
+        output_var_name: Optional[str] = None,
+        skip_remove_unused_variables: bool = False,
+    ) -> str:
+        """
+        Generate code as string using list of stored statements
+
+        Parameters
+        ----------
+        to_format: bool
+            Whether to format the final code
+        output_var_name: Optional[str]
+            Output variable name, default to "output"
+        skip_remove_unused_variables: bool
+            Whether to skip removing unused variables
+
+        Returns
+        -------
+        str
+        """
+        # generate statements and extract required imports
+        statements, import_pairs = self._generate()
+
+        if not skip_remove_unused_variables:
+            # extract unused variables
+            tree = ast.parse(statements)
+            finder = UnusedVariableFinder()
+            finder.visit(tree)
+            keep_var_names = {output_var_name} if output_var_name else {"output"}
+            unused_variables = finder.get_unused_variables().difference(keep_var_names)
+
+            # regenerate statements by removing unused variables
+            statements, import_pairs = self._generate(unused_variables=unused_variables)
```

Comment:
Is it guaranteed that in the regeneration, the variable names will be generated in the exact same way?

Reply:
Yes. Since the variable name has been generated. The main side effect is that we may have variable name with skip numbering. If we have variables like `col_1`, `col_2`, `col_3`, if `col_2` is unused, the variable used will be `col_1` and `col_3`.